internal/service/ipam/api_subnet_resource_test.go

```diff
@@ -1059,6 +1090,77 @@ resource "bloxone_ipam_subnet" "test_ddns_conflict_resolution_mode" {
 	return strings.Join([]string{testAccBaseWithIPSpace(), config}, "")
 }
 
+func testAccSubnetDhcpOptionsOption(address string, cidr int, name, type_, optValue string) string {
+	config := fmt.Sprintf(`
+resource "bloxone_ipam_subnet" "test_dhcp_options" {
+	address = %q
+    cidr = %d
+    space = bloxone_ipam_ip_space.test.id
+    name = %q
+    dhcp_options = [
+      {
+       type = %q
+       option_code = bloxone_dhcp_option_code.test.id
+       option_value = %q
+      }
+    ]
+}
+`, address, cidr, name, type_, optValue)
+
+	return strings.Join([]string{testAccBaseWithIPSpace(), testAccOptionBasicConfig(), config}, "")
+}
+
+func testAccSubnetDhcpOptionsGroup(address string, cidr int, name, type_ string) string {
+	config := fmt.Sprintf(`
+resource "bloxone_ipam_subnet" "test_dhcp_options" {
+	address = %q
+    cidr = %d
+    space = bloxone_ipam_ip_space.test.id
+    name = %q
+    dhcp_options = [
+      {
+       type = %q
+       group = bloxone_dhcp_option_group.test.id
+      }
+    ]
+}
+`, address, cidr, name, type_)
+
+	return strings.Join([]string{testAccBaseWithIPSpace(), testAccOptionBasicConfig(), config}, "")
+}
+
+func testAccOptionBasicConfig() string {
```

Comment:
```suggestion
func testAccBaseWithOptionSpaceAndCode() string {
```

Reply:
If this function is already used in the basic test, do not reuse it, create another function with the above name.

Reply:
This is a new function created. renamed it